searchcore/src/vespa/searchcore/proton/bucketdb/bucketdb.h

```diff
@@ -21,12 +22,21 @@ class BucketDB
     using BucketState = bucketdb::BucketState;
     using Map = vespalib::hash_map<BucketId, BucketState, document::BucketId::hash>;
 
-    Map _map;
-    BucketId _cachedBucketId;
-    BucketState _cachedBucketState;
+    Map                  _map;
+    std::atomic<size_t>  _numActiveDocs;
+    BucketId             _cachedBucketId;
+    BucketState          _cachedBucketState;
 
     void clear();
     void checkEmpty() const;
+    size_t countActiveDocs() const;
+    void checkActiveCount() const;
+    void addActive(size_t value) {
+        _numActiveDocs.fetch_add(value, std::memory_order_relaxed);
+    }
+    void subActive(size_t value) {
+        _numActiveDocs.fetch_sub(value, std::memory_order_relaxed);
```

Comment:
If these values are only ever mutated from a single writer, the explicit `fetch_add`/`fetch_sub` could potentially be replaced with `store(load(relaxed) +/- 1, relaxed)` to avoid a couple of locked instructions

Reply:
Fixed